api/src/main/java/bio/terra/tanagra/service/query/api/ConversionUtils.java

```diff
@@ -0,0 +1,23 @@
+package bio.terra.tanagra.service.query.api;
+
+import bio.terra.common.exception.BadRequestException;
+import bio.terra.tanagra.service.search.Variable;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+/** Utilities for converting from API model classes to Tanagra classes. */
+public final class ConversionUtils {
+  private ConversionUtils() {}
+
+  /** Returns if exactly one of the objects is non-null. */
+  public static boolean exactlyOneNonNull(Object... objects) {
+    return Stream.of(objects).filter(Objects::nonNull).count() == 1;
+  }
+
+  public static Variable createVariable(String name) {
+    if (name.isEmpty()) {
+      throw new BadRequestException("A variable must have a non-empty name.");
+    }
+    return Variable.create(name);
```

Comment:
nit: why this utility, rather than just adding this validation to `Variable.create` ? Reading further, it looks like you also did that. Do you need this util then?

Reply:
Got it. I think that distinction is probably correct - I just missed it. It's correct for it to come from the conversion layer.